sources/directconnect/direct-connect-gateway-attachment.go

```diff
@@ -0,0 +1,121 @@
+			// +overmind:link directconnect-direct-connect-gateway
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-direct-connect-gateway",
+					Method: sdp.QueryMethod_GET,
+					Query:  *attachment.DirectConnectGatewayId,
+					Scope:  "global",
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// This is not clearly documented, but it seems that if the gateway is deleted, the attachment state will change to detaching
+					In: true,
+					// We can't affect the direct connect gateway
+					Out: false,
+				},
+			})
+		}
+
+		if attachment.VirtualInterfaceId != nil {
+			// +overmind:link directconnect-virtual-interface
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-virtual-interface",
+					Method: sdp.QueryMethod_GET,
+					Query:  *attachment.VirtualInterfaceId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// If virtual interface is deleted, the attachment state will change to detaching
+					// https://docs.aws.amazon.com/directconnect/latest/APIReference/API_DirectConnectGatewayAttachment.html#API_DirectConnectGatewayAttachment_Contents
+					In: true,
+					// We can't affect the virtual interface
+					Out: false,
+				},
+			})
+		}
+
+		items = append(items, &item)
+	}
+
+	return items, nil
+}
+
+//go:generate docgen ../../docs-data
+// +overmind:type directconnect-direct-connect-gateway-attachment
+// +overmind:descriptiveType Direct Connect Gateway Attachment
+// +overmind:get Get a direct connect gateway attachment by DirectConnectGatewayId and VirtualInterfaceId
+// +overmind:search Search direct connect gateway attachments for given VirtualInterfaceId
+// +overmind:group AWS
+
+func NewDirectConnectGatewayAttachmentSource(config aws.Config, accountID string, limit *sources.LimitBucket) *sources.DescribeOnlySource[*directconnect.DescribeDirectConnectGatewayAttachmentsInput, *directconnect.DescribeDirectConnectGatewayAttachmentsOutput, *directconnect.Client, *directconnect.Options] {
+	return &sources.DescribeOnlySource[*directconnect.DescribeDirectConnectGatewayAttachmentsInput, *directconnect.DescribeDirectConnectGatewayAttachmentsOutput, *directconnect.Client, *directconnect.Options]{
+		Config:    config,
+		Client:    directconnect.NewFromConfig(config),
+		AccountID: accountID,
+		ItemType:  "directconnect-direct-connect-gateway-attachment",
+		DescribeFunc: func(ctx context.Context, client *directconnect.Client, input *directconnect.DescribeDirectConnectGatewayAttachmentsInput) (*directconnect.DescribeDirectConnectGatewayAttachmentsOutput, error) {
+			limit.Wait(ctx) // Wait for rate limiting
+			return client.DescribeDirectConnectGatewayAttachments(ctx, input)
+		},
+		InputMapperGet: func(scope, query string) (*directconnect.DescribeDirectConnectGatewayAttachmentsInput, error) {
```

Comment:
This source does not support List All operation without giving any parameters.

https://awscli.amazonaws.com/v2/documentation/api/latest/reference/directconnect/describe-direct-connect-gateway-attachments.html